src/plots/cartesian/axes.js

```diff
@@ -2968,20 +2968,41 @@ axes.makeTransTickFn = function(ax) {
 
 axes.makeTransTickLabelFn = function(ax) {
     var uv = getTickLabelUV(ax);
+    var shift = ax.ticklabelshift || 0;
+    var standoff = ax.ticklabelstandoff || 0;
+
     var u = uv[0];
     var v = uv[1];
 
-    return ax._id.charAt(0) === 'x' ?
+    var isXaxis = ax._id.charAt(0) === 'x';
+    var isYaxis = !isXaxis;
+    var isReversed = ax.range[0] > ax.range[1];
+    var labelsInside = ax.ticklabelposition && ax.ticklabelposition.indexOf('inside') !== -1;
+    var labelsOutside = !labelsInside;
+
+    if(shift) {
+        var shiftSign = isReversed ? -1 : 1;
+        shift = shift * shiftSign;
+    }
+    if(standoff) {
+        var standoffSign =
+            isXaxis && ax.side === 'bottom' && labelsOutside ||
+            isXaxis && ax.side === 'top' && labelsInside ||
+            isYaxis && ax.side === 'right' && labelsOutside ||
+            isYaxis && ax.side === 'left' && labelsInside ? 1 : -1;
```

Comment:
```suggestion
        var standoffSign = (
            (isXaxis && ax.side === 'bottom' && labelsOutside) ||
            (isXaxis && ax.side === 'top' && labelsInside) ||
            (isYaxis && ax.side === 'right' && labelsOutside) ||
            (isYaxis && ax.side === 'left' && labelsInside)
        ) ? 1 : -1;
```

Reply:
Or alternatively 
```js
        var side = ax.side;
        var standoffSign = (
            (labelsInside && (side === 'top' || side === 'left')) ||
            (labelsOutside && (side === 'bottom' || side === 'right'))
        ) ? 1 : -1;
```

Reply:
Right, I forgot that axis/side mismatches are cleaned before 👍

Reply:
Btw, I noticed at several locations in the plotly code that the properties of objects are often assigned to a variable and then the variable is used instead of using the property directly, like you do above in your suggestion. Is there a reason for this?

Reply:
Good question.
IMHO, when using an identical attribute several times in one part of code, there is a good chance that one makes a typo and it is sometimes hard to find & fix the bug.
In addition this could help improve the performance (of look up for the value) and reduce the minified bundle size as the long attribute names just appear once.